### PR TITLE
[FEAT][C++] Change the default namespace to `graphar`

### PR DESCRIPTION
**Pull request by acezen**
## Proposed changes
as #412 describe

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/alibaba/GraphAr/blob/main/CONTRIBUTING.rst) doc
- [ ] I have signed the CLA
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments
close issue #412

### CODE REVIEW

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -37,15 +37,15 @@ project(graph-archive LANGUAGES C CXX VERSION ${GAR_VERSION})
...
 
-option(NAMESPACE "User specific namespace, default if GraphArchive" OFF)
+option(NAMESPACE "User specific namespace, default is graphar" OFF)
 option(BUILD_TESTS "Build unit tests" OFF)
 option(BUILD_EXAMPLES "Build examples" OFF)
 option(BUILD_BENCHMARKS "Build benchmarks" OFF)
 
 if (NAMESPACE)
-  add_definitions(-DGAR_NAMESPACE=${NAMESPACE})
+  add_definitions(-DNAMESPACE=${NAMESPACE})
```
Comment (lixueclaire) [inline cpp/CMakeLists.txt:46]: DGAR_NAMESPACE

File: cpp/CMakeLists.txt [hunk 2]
```diff
@@ -37,15 +37,15 @@ project(graph-archive LANGUAGES C CXX VERSION ${GAR_VERSION})
...
 else()
-  add_definitions(-DGAR_NAMESPACE=GraphArchive)
+  add_definitions(-DNAMESPACE=graphar)
```
Comment (lixueclaire) [inline cpp/CMakeLists.txt:48]: ditto

File: cpp/include/gar/util/macros.h [hunk 3]
```diff
@@ -38,7 +38,7 @@
 #if defined(GAR_NAMESPACE)
 #define GAR_NAMESPACE_INTERNAL GAR_NAMESPACE
```
Comment (lixueclaire) [inline cpp/include/gar/util/macros.h:39]: we use `GAR_NAMESPACE` here

File: cpp/CMakeLists.txt (hunk 2 above)
Comment (acezen) [reply cpp/CMakeLists.txt]: Fixed

File: cpp/CMakeLists.txt (hunk 1 above)
Comment (acezen) [reply cpp/CMakeLists.txt]: Fixed

**Comment by acezen** [comment]
The upstream arrow's debain package seems break https://github.com/apache/arrow/issues/40759

**Review by lixueclaire — APPROVED** [review]
LGTM

**Pull request merged**